packages/help-center/src/hooks/use-should-use-wapuu.ts

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-restricted-imports */
+import config from '@automattic/calypso-config';
+import { useSupportAvailability } from '../data/use-support-availability';
+
+function isWapuuFlagSetInURL(): boolean {
+	const currentUrl = window.location.href;
+	const urlParams = new URLSearchParams( new URL( currentUrl ).search );
+	return urlParams.get( 'flags' ) === 'wapuu';
+}
+
+export const useShouldUseWapuu = () => {
+	const { data: supportAvailability } = useSupportAvailability();
+
+	// All users eligible for support should see the Wapuu assistant.
+	const isEligibleForSupport = Boolean( supportAvailability?.is_user_eligible );
+
+	// Force Wapuu via URL flag
+	const isFlagSetInURL = isWapuuFlagSetInURL();
+
+	// Wapuu can be enabled via config
+	const isConfigEnabled = config.isEnabled( 'wapuu' );
```

Comment:
Might not be true. Maybe I'm wrong, but I had to add that isFlagSetInUrl due to `wp-admin` . I think for `Calypso` your reasoning is correct. But when we need to load via flags from `wp-admin` it does not work.

Please, double check, maybe I'm wrong :)

Reply:
**TIL**

<img width="293" alt="Markup 2024-06-18 at 11 41 46" src="https://github.com/Automattic/wp-calypso/assets/33258733/f6e2c64c-e4df-43e8-9c18-693321d7c01a">

Reply:
But does it work if used within `wp-admin`?

Reply:
Works for me 
![image](https://github.com/Automattic/wp-calypso/assets/17054134/9a977dd4-4e04-489b-aa74-76940fc85ebd)

Reply:
OK that only worked for me because I'm running `yarn dev --sync`, i.e my NODE_ENV is `development`. It won't work in prod. You are right. 

See the checks [here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-config/src/index.ts#L87-L89).

Reply:
Thanks for the checking, I got lost on another topic and forgot about this thread!